File: frameworks/PHP/dreamcatframework/src/main/View/FortuneView.php
<?php

namespace DreamCat\Benchmark\View;

use DreamCat\Benchmark\Entry\Fortune\FortuneEntry;

/**
 * Fortune的视图层
 * @author vijay
 */
class FortuneView
{
    /**
     * 生成message的html
     * @param FortuneEntry[] $list
     * @return string
     */
    public function messageHtml(array $list): string
    {
        $html = implode("", array_map(function (FortuneEntry $entry) {
            $msg = htmlspecialchars($entry->getMessage(), ENT_QUOTES, "UTF-8");
            return "<tr><td>{$entry->getId()}</td><td>{$msg}</td></tr>";
        }, $list));
        return "<!DOCTYPE html><html><head><title>Fortunes</title></head><body><table><tr><th>id</th>"
            . "<th>message</th></tr>{$html}</table></body></html>";
    }
}

# end of file

File: frameworks/Nim/basolato.bk/resources/pages/fortune_view.nim
import json
import basolato/view

proc impl(title:string, data:seq[JsonNode]):string = tmpli html"""
<!DOCTYPE html>
<html>

<head>
  <title>$title</title>
</head>

<body>
  <table>
    <tr>
      <th>id</th>
      <th>message</th>
    </tr>
    $for row in data{
      <tr>
        <td>$(row["id"].get)</td>
        <td>$(row["message"].get)</td>
      </tr>
    }
  </table>
</body>

</html>
"""

proc fortuneView*(this:View, data=newSeq[JsonNode]()):string =
  let title = "Fortunes"
  return impl(title, data)
